userScripts documentation fixups (#38348)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/userscripts/index.md

When a user script is registered or updated (using {{WebExtAPIRef("userScripts.register()")}} or {{WebExtAPIRef("userScripts.update()")}}), your extension can set it to run in an isolated `USER_SCRIPT` world or the `MAIN` world.

> User scripts can share the same world, but they can also be isolated from each other by declaring separate `USER_SCRIPT` worlds through the `worldId` property of a {{WebExtAPIRef("userScripts.RegisteredUserScript", "RegisteredUserScript")}}. The API also enables an extension to configure a content security policy (CSP) for the `USER_SCRIPT` world using {{WebExtAPIRef("userScripts.configureWorld()")}}.

Reviewer: ```suggestion
User scripts can share a `USER_SCRIPT` world or isolate themselves by setting the `worldId` property of {{WebExtAPIRef("userScripts.RegisteredUserScript", "RegisteredUserScript")}}. The API enables an extension to configure a content security policy (CSP) for a `USER_SCRIPT` world using {{WebExtAPIRef("userScripts.configureWorld()")}}.
```

> A `USER_SCRIPT` world provides an isolated execution environment that isn't accessible to a host page or other extensions. This means a user script can change its JavaScript environment without affecting the host page or other extensions' user and content scripts. In this environment, user scripts aren't visible to the host page or other extensions' user and content scripts.

Reviewer: ```suggestion
A `USER_SCRIPT` world provides an isolated execution environment that isn't accessible to a host page or other extensions. This means a user script isn't visible to and can change its JavaScript environment without affecting the host page or other extensions' user and content scripts.
```

Author: This proposed text doesn't look very readable, "isn't visible to and" looks strange.

I think that a follow-up improvement could be to just cut most of the text and refer to https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/userScripts/ExecutionWorld instead for a longer description

Reviewer: I want to avoid readability issues created by two apparently very similar sentences. How about this?

```suggestion
A `USER_SCRIPT` world provides an isolated execution environment that isn't accessible to a host page or other extensions. This means a user script can change its JavaScript environment without affecting and isn't visible to its host page or other extensions' user and content scripts.
```

Author: How about this? And maybe switch up with the previous paragraph?

```suggestion
A `USER_SCRIPT` world provides an isolated execution environment that isn't accessible to a host page or other extensions. This isolation is similar to a [content script environment](/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts#content_script_environment), except `USER_SCRIPT` worlds cannot access extension APIs.
```

> In the `MAIN` world, host pages and other extensions can see and access running user scripts. The `worldId` property is not supported for `MAIN` worlds.

Author: The "`USER_SCRIPT` worlds are created by setting the `worldId` property" bit is inaccurate. What `worldId` does is allowing for separate/multiple USER_SCRIPT worlds to be specified. It can be omitted to still have the (default) USER_SCRIPT world.

> In the `MAIN` world, host pages and other extensions can see and access running user scripts. The `worldId` property is not supported for `MAIN` worlds.

Author: The suggested text does not convey the intended meaning; To avoid unnecessary delays in documentation I'm going to merge this PR. Feel free to submit a new PR to improve the wording if you'd like.
